sbtc-core/src/signer/mod.rs

```diff
@@ -39,6 +45,12 @@ pub trait Sign {
     fn verify_message(&self, public_key: &ecdsa::PublicKey, message: &[u8]) -> SBTCResult<bool>;
 }
 
+/// Validator trait for validating pending Bitcoin transactions
+pub trait Validator {
+    /// Validate the given signable Bitcoin transaction
+    fn validate_transaction(&self, tx: &SignableTransaction) -> SBTCResult<bool>;
+}
+
 /// sBTC compliant Signer
 pub struct Signer<S: Sign + Coordinate> {
```

Comment:
Although unrelated to this PR, I realized it's bad practice to add trait bounds to structs - since structs are about data organization, while traits are about behavior. Best practices is to just omit the trait bounds in the struct definition, and use them in the impl blocks.

```suggestion
pub struct Signer<S> {
```

Reply:
I did it this way to ensure if a user wants to create their own signer using the SDK, they cannot without implementing these traits. But can def pull it out.